### DISCUSSION TITLE
[VTS] Non active validators won't have size score

### DISCUSSION THREAD

**Discussion by Albermonte** [nimiq/developer-center#25] [General]
From the docs:

> Due to technical limitations, we can currently only calculate the size of validators that are active in the current epoch. We cannot calculate the score of a validator at a given timestamp.

This means that if the validator is not active in the last epoch, it's score would be 0, even though it should only be penalized a bit in the liveness.

## Workaround
An alternative solution would be to use the current stake of the validator as the size, since that's the closet to the last finished epoch.

**Comment by sisou** [comment]
The size can be calculated in two ways:
1. Current stake vs total stake
2. Currently assigned slots vs all slots (512)

The first way can only be done for the current time, no matter how far into an epoch we already are.
The second way can only be done for elections (and the most recent election being the relevant one).

I think we should calculate both ratios, and take the bigger one for the score. That would prevent a validator from only adding a huge stake right before an election block and then removing it again, for a (currently) better score. Get the slots, but not the size penalty. 

On the other hand, if a validator was very small at the last election (good score), but has received lots of stake in the meantime (for the next election), we should already disincentivise people from staking with it.

So both scores are needed and the worse one for the validator must be used.
